src/ensembl/src/shared/components/species-tabs-wrapper/SingleLineSpeciesWrapper.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { ReactElement, useState, useEffect, useRef } from 'react';
+import useResizeObserver from 'use-resize-observer';
+import { useSprings, animated } from 'react-spring';
+
+import { getSpeciesItemWidths } from './speciesTabsWrapperHelpers';
+
+import styles from './SingleLineSpeciesWrapper.scss';
+
+import { Props as FocusableSelectedSpeciesProps } from 'src/shared/components/selected-species/FocusableSelectedSpecies';
+
+export type Props = {
+  isWrappable: false;
+  speciesTabs: ReactElement<FocusableSelectedSpeciesProps>[];
+  link?: React.ReactNode | null;
+};
+
+const animationCalculator = ({
+  items,
+  hoveredItemIndex,
+  containerRef,
+  containerWidth,
+  immediate = false
+}: {
+  items: FocusableSelectedSpeciesProps[];
+  hoveredItemIndex: number | null;
+  containerRef: React.RefObject<HTMLElement>;
+  containerWidth: number;
+  immediate: boolean;
+}) => {
+  const springConfig = { tension: 280, friction: 45 };
+  const updatedParamsList = items.map((item, index) => ({
+    ...item,
+    isHovered: index === hoveredItemIndex
+  }));
+  if (containerRef.current) {
+    return getSpeciesItemWidths({
+      items: updatedParamsList,
+      containerWidth
+    }).map((width) => ({
+      config: springConfig,
+      width: `${width}px`,
+      immediate
+    }));
+  } else {
+    return items.map(() => ({
+      config: springConfig,
+      width: '0px',
+      immediate
```

Comment:
Any change `springConfig` and `immediate` can be moved out of here as they are going to be the same for all items?

Reply:
Actually, after some experimentation, I decided to replace react-spring in this component with native CSS transition — it definitely performs better.